res/controllers/Numark-Mixtrack-Pro-FX-scripts.js

```diff
@@ -0,0 +1,792 @@
+var MixtrackProFX = {};
+
+// pitch ranges
+// add/remove/modify steps to your liking
+// default step must be set in Mixxx settings
+// setting is stored per deck in pitchRange.currentRangeIdx
+MixtrackProFX.pitchRanges = [0.08, 0.16, 1];
+
+// whether the corresponding Mixxx option is enabled
+// (Settings -> Preferences -> Waveforms -> Synchronize zoom level across all waveforms)
+MixtrackProFX.waveformsSynced = true;
+
+// jogwheel
+MixtrackProFX.jogScratchSensitivity = 1024;
+MixtrackProFX.jogScratchAlpha = 1; // do NOT set to 2 or higher
+MixtrackProFX.jogScratchBeta = 1/32;
+MixtrackProFX.jogPitchSensitivity = 10;
+MixtrackProFX.jogSeekSensitivity = 10000;
+
+// blink settings
+MixtrackProFX.enableBlink = true;
+MixtrackProFX.blinkDelay = 700;
+
+// autoloop sizes, for available values see:
+// https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-beatloop_X_toggle
+MixtrackProFX.autoLoopSizes = [
+    "0.0625",
+    "0.125",
+    "0.25",
+    "0.5",
+    "1",
+    "2",
+    "4",
+    "8"
+];
+
+// beatjump values, for available values see:
+// https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-beatjump_X_forward
+// underscores (_) at the end are needed because numeric values (e.g. 8) have two underscores (e.g. beatjump_8_forward),
+// but "beatjump_forward"/"beatjump_backward" have only one underscore
+MixtrackProFX.beatJumpValues = [
+    "0.0625_",
+    "0.125_",
+    "0.25_",
+    "0.5_",
+    "1_",
+    "2_",
+    "", // "beatjump_forward"/"beatjump_backward" - jump by the value selected in Mixxx GUI (4 by default)
+    "8_"
+];
+
+// dim all lights when inactive instead of turning them off
+components.Button.prototype.off = 0x01;
+
+// pad modes control codes
+MixtrackProFX.PadModeControls = {
+    HOTCUE: 0x00,
+    AUTOLOOP: 0x0D,
+    FADERCUTS: 0x07,
+    SAMPLE1: 0x0B,
+    BEATJUMP: 0x02,
+    SAMPLE2: 0x0F
+};
+
+// state variable, don't touch
+MixtrackProFX.shifted = false;
+
+MixtrackProFX.init = function() {
+    // send sysexes
+    var exitDemoSysex = [0xF0, 0x7E, 0x00, 0x06, 0x01, 0xF7];
+    midi.sendSysexMsg(exitDemoSysex, exitDemoSysex.length);
+
+    var statusSysex = [0xF0, 0x00, 0x20, 0x7F, 0x03, 0x01, 0xF7];
+    midi.sendSysexMsg(statusSysex, statusSysex.length);
+
```

Comment:
Sending the status query sysex here does not make much sense because none of the receiving handlers have been created yet.

Reply:
Can you put this part below the deck and effect creation section?